packages/trader/src/Modules/Contract/Components/LastDigitPrediction/digit-display.tsx

```diff
@@ -1,14 +1,39 @@
 import classNames from 'classnames';
 import { observer } from 'mobx-react';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { DesktopWrapper } from '@deriv/components';
 import { isMobile } from '@deriv/shared';
 import { Bounce } from 'App/Components/Animations';
-import Digit from './digit.jsx';
-import DigitSpot from './digit-spot.jsx';
-import LastDigitStat from './last-digit-stat.jsx';
+import Digit from './digit';
+import DigitSpot from './digit-spot';
+import LastDigitStat from './last-digit-stat';
+import { ProposalOpenContract } from '@deriv/api-types';
 
+type TDigitDisplay = {
+    barrier: number | null;
+    is_digit_contract?: boolean;
+    has_entry_spot: boolean;
+    is_lost: React.ComponentProps<typeof Digit>['is_lost'] | React.ComponentProps<typeof DigitSpot>['is_lost'];
```

Comment:
Just asking: isn't it boolean? Please, ignore if not relevant

Reply:
Same as above. Defined in child component as boolean. Being more explicit/typesafe rather than giving it a primitive data type. That being said `is_lost = boolean` defined this way.

Reply:
Data type will follow where ever this component is reused.